ee/clickhouse/queries/funnels/funnel_trends_persons.py

```diff
@@ -1,6 +1,51 @@
-from ee.clickhouse.queries.funnels.base import ClickhouseFunnelBase
+from datetime import datetime
 
+from ee.clickhouse.queries.funnels.funnel_trends import TIMESTAMP_FORMAT, ClickhouseFunnelTrends
+from ee.clickhouse.queries.util import get_trunc_func_ch
+from posthog.models.person import Person
 
-class ClickhouseFunnelTrendsPersons(ClickhouseFunnelBase):
-    def get_query(self):
-        pass
+
+class ClickhouseFunnelTrendsPersons(ClickhouseFunnelTrends):
+    PAGE_SIZE = 100
+
+    def get_query(self) -> str:
```

Comment:
passing comment: it's ideal if `ClickhouseFunnelTrends` itself also implements `get_steps_per_person` and `get_steps_per_person_without_aggregation`, which you can then use immediately in `TrendsPersons`.

This also helps us make `get_steps_per_person*` abstract methods in the base class, such that we expect every funnel type to implement them :)

I'm not sure how `get_steps_per_person` would look like, since the "steps" become time boxes in trends, but pretty sure we can have the `get_steps_per_person_without_aggregation` bit implemented.

Reply:
Hm, "use immediately" to what extent?

Reply:
Sorry, should've been clearer. For example, instead of getting the `steps_per_person_query` from **funnel_order** class, you'd instead do:

```
steps_per_person_query = self.get_step_counts_without_aggregation_query()
```

and this function would be the query:

```
SELECT
                        person_id,
                        {interval_method}(timestamp) AS entrance_period_start,
                        max(steps) AS steps_completed
                    FROM (
                        {steps_per_person_query}
                    ) GROUP BY person_id, entrance_period_start
```

which becomes the base query for use in funnelTrends `get_query` and in persons.

(ofcourse, disregard if this doesn't work out well, but I think it does, and then, you can reuse the pattern everywhere else, like so: https://github.com/PostHog/posthog/blob/master/ee/clickhouse/queries/funnels/funnel_unordered_persons.py#L6-L12

I think the HAVING clause can go into the where if you turn it into the subquery, and *I think* the performance impact here isn't significant enough to trade-off against the common pattern we have for person queries.